lib/src/chart/scatter_chart/scatter_chart_data.dart

```diff
@@ -571,3 +578,76 @@ class ScatterChartDataTween extends Tween<ScatterChartData> {
     return begin!.lerp(begin!, end!, t);
   }
 }
+
+/// It gives you the index value of the spot and gets the text style of the label.
+typedef GetLabelTextStyleFunction = TextStyle Function(int spotIndex);
+
+/// It gives you the index value of the spot and returns the label of the spot.
+typedef GetLabelFunction = String Function(int spotIndex);
+
+/// It gives you the default text style of the label for a spot.
+TextStyle getDefaultLabelTextStyleFunction(int spotIndex) {
+  return const TextStyle();
```

Comment:
You can make it nullable and return `null` here.
Then we use a textStyle from the parent theme.

Reply:
> It doesn't work good on our ScatterChartSample2 with this code:
> 
> ```dart
> scatterLabelSettings: ScatterLabelSettings(
>   showLabel: true,
>   getLabelFunction: (index) => index.toString(),
>   getLabelTextStyleFunction: (index) => const TextStyle(
>     color: Colors.white,
>     fontSize: 40,
>   ),
> ),
> ```
> 
> <img alt="image" width="393" src="https://user-images.githubusercontent.com/7009300/155397943-fef773e4-1752-49b2-9ce7-c20fa230c29d.png">

@imaNNeoFighT  I think this is because the size of the label is not getting adjusted to the window size. I believe so. What do you think?

Reply:
it needs to be scaled with the canvas size or the window size. I saw this issue with the axis titles as well, but it is outside the chart. So it doesn't affect much.